docs/RUN.md

```diff
@@ -19,6 +19,47 @@ Now that gdbgui is open, you can interactively run a program with it.
 
 For a list of gdbgui arguments, see the Arguments section below or type gdbgui --help.
 
+### Debugging Rust Programs
+
+`gdbgui` can be used to debug programs written in Rust. Assuming you use [Cargo](https://doc.rust-lang.org/stable/cargo/) to create a new program
+and build it in Debug mode in the standard way:
+
+```
+cargo new myprog
+cd myprog
+cargo build
+```
+
+You can start debugging with a simple
+
+`gdbgui target/debug/myprog`
+
+There are a couple of small niggles.
+
+Instead of showing your `main` function the initial screen will be blank and `gdbgui` will print `File not found: main`.
+You need to help `gdbgui` out by typing `main` into the file browser box:
+
+![](https://raw.githubusercontent.com/cs01/gdbgui/master/screenshots/rust_main.png)
+
+and selecting the `main.rs` file. The source code should then appear in the browser and you can click to set breakpoints
+and run the program. Of course, if you want to break in some other file, you can find that in the file browser instead.
+
+The second niggle is not with `gdbgui` but with the GDB pretty-printing macros that Rust ships with. GDB can't find these by default,
```

Comment:
This implies the first has something to do with gdbgui, but I don't think it does. It has to do with gdb's response to the `break main` command.

gdbgui runs the gdb command `-break-insert main`, and then gets a breakpoint back from gdb. The breakpoint has a field called fullfile which is usually a path to a file, but when debugging rust, gdb cannot find the function and/or file and it just puts a breakpoint at an address.